MaterialDesignThemes.Wpf/Themes/MaterialDesignTheme.TextBox.xaml

```diff
@@ -16,7 +16,7 @@
         <Setter Property="BorderThickness" Value="0 0 0 1"/>
         <Setter Property="wpf:TextFieldAssist.TextBoxViewMargin" Value="0 0 1 0" />
         <Setter Property="Background" Value="Transparent"/>
-        <Setter Property="CaretBrush" Value="{DynamicResource PrimaryHueMidBrush}"/>
+        <Setter Property="CaretBrush" Value="{Binding Path=(wpf:TextFieldAssist.UnderlineHighlightedBrush), RelativeSource={RelativeSource Self}}" />
```

Comment:
I would not re-use the underline brush here and simply leave it as `{DynamicResource PrimaryHueMidBrush}`